### PR TITLE
feat: opt-in language exp mp-270

### PR DESCRIPTION
**Pull request by christian14b**
Checkout and Register Social

Experiment:
image

### CODE REVIEW

File: src/components/Checkout/CheckoutDropInPaymentWrapper.vue [hunk 1]
```diff
@@ -77,11 +77,11 @@
...
 						:href="`https://${this.$appConfig.host}/legal/privacy`"
 						target="_blank"
-						title="Open Privacy Policy in a new window"
-					>Privacy Policy</a>.
+						:title="`Open Privacy ${enableCommsExperiment ? 'Notice' : 'Policy' } in a new window`"
+					>Privacy {{ enableCommsExperiment ? 'Notice' : 'Policy' }}</a>.
 					<p v-if="$v.termsAgreement.$error" class="input-error tw-text-danger tw-text-base">
 						You must agree to the Kiva Terms of service & Privacy
-						policy.
+						{{ enableCommsExperiment ? 'Notice' : 'Policy' }}.
```
Comment (dyersituations) [inline src/components/Checkout/CheckoutDropInPaymentWrapper.vue:84]: It looks like "policy" was lowercase before, is having it always uppercase expected?

File: src/components/Checkout/CheckoutDropInPaymentWrapper.vue [hunk 2]
```diff
@@ -148,6 +152,11 @@ export default {
 		KvTextInput,
 	},
 	inject: ['apollo', 'cookieStore'],
+	apollo: {
```
Comment (dyersituations) [inline src/components/Checkout/CheckoutDropInPaymentWrapper.vue:155]: This should already be handled globally here: https://github.com/kiva/ui/blob/4085cf13460d40fc9771fd793128d222847b246a/src/util/experiment/experimentUtils.js#L343

Does this still work like expected with this `preFetch` removed?

File: src/pages/LoginAndRegister/RegisterSocial.vue [hunk 3]
```diff
@@ -198,6 +207,16 @@ export default {
 			const inner = parts.length ? `${parts.join(', ')} and ${last}` : last;
 			return `To finish creating your account, please ${inner} below.`;
 		},
+		emailUpdatesCopy() {
+			if (this.enableCommsExperiment) {
```
Comment (dyersituations) [inline src/pages/LoginAndRegister/RegisterSocial.vue:211]: Will we have a situation where `passwordless` and `enableCommsExperiment` are both enabled?

File: src/pages/LoginAndRegister/RegisterSocial.vue [hunk 4]
```diff
@@ -258,18 +277,41 @@ export default {
...
-				query: strategicPartnerLoginInfoByPageIdQuery,
-				variables: { pageId: route.query.partnerContentId ?? '' }
-			});
-		},
+			return Promise.all([
+				client.query({
+					query: strategicPartnerLoginInfoByPageIdQuery,
+					variables: { pageId: route.query.partnerContentId ?? '' }
+				}),
+				client.query({
```
Comment (dyersituations) [inline src/pages/LoginAndRegister/RegisterSocial.vue:309]: See other comment, this probably isn't needed.

File: src/pages/LoginAndRegister/RegisterSocial.vue (hunk 3 above)
Comment (mcstover) [reply src/pages/LoginAndRegister/RegisterSocial.vue:211]: I would say if `passwordless` is active the exp should be bypassed. The corporate partner context isn't something we want to "mix" with general experimentation and passwordless is specifically for that context at this time.

File: src/components/Checkout/CheckoutDropInPaymentWrapper.vue [hunk 5]
```diff
@@ -77,11 +77,11 @@
...
 						:href="`https://${this.$appConfig.host}/legal/privacy`"
 						target="_blank"
-						title="Open Privacy Policy in a new window"
-					>Privacy Policy</a>.
+						:title="`Open Privacy ${enableCommsExperiment ? 'Notice' : 'Policy' } in a new window`"
+					>Privacy {{ enableCommsExperiment ? 'Notice' : 'Policy' }}</a>.
 					<p v-if="$v.termsAgreement.$error" class="input-error tw-text-danger tw-text-base">
 						You must agree to the Kiva Terms of service & Privacy
-						policy.
+						{{ enableCommsExperiment ? 'notice' : 'policy' }}.
```
Comment (dyersituations) [inline src/components/Checkout/CheckoutDropInPaymentWrapper.vue:84]: I see this was reverted to be lowercase, but do we actually want it uppercase to match other usages in the page?

File: src/components/WwwFrame/TheHeader.vue [hunk 6]
```diff
@@ -798,6 +800,17 @@ export default {
 			}
 		}
 
+		if (this.cookieStore.get(COMMS_OPT_IN_EXP_KEY) !== 'true') {
```
Comment (dyersituations) [inline src/components/WwwFrame/TheHeader.vue:803]: Why check the cookie value here?

File: src/pages/LoginAndRegister/RegisterSocial.vue (hunk 3 above)
Comment (dyersituations) [reply src/pages/LoginAndRegister/RegisterSocial.vue:210]: It looks like this still needs to be bypassed if `passwordless` is `true`

**Review by dyersituations — APPROVED** [review]

**Pull request merged**